pkg/kubelet/kubelet_pods.go

```diff
@@ -1885,6 +1886,13 @@ func (kl *Kubelet) convertToAPIContainerStatuses(pod *v1.Pod, podStatus *kubecon
 			ImageID:      cs.ImageID,
 			ContainerID:  cid,
 		}
+		if !cs.StartedAt.IsZero() && cs.FinishedAt.IsZero() && !hasStartupProbe {
+			// without a startup probe, mark this container started during the time it is running
+			status.Started = pointer.Bool(true)
+		} else {
+			// with a startup probe, prober_manager will set the started field depending on the probe result
+			status.Started = pointer.Bool(false)
+		}
```

Comment:
nit:
```suggestion
                // without a startup probe, mark this container started during the time it is running
                // otherwise set the started field depending on the probe result
		status.Started = pointer.Bool(!cs.StartedAt.IsZero() && cs.FinishedAt.IsZero() && !hasStartupProbe) {
```